lib/Watcher.js

```diff
@@ -66,34 +67,47 @@ export default class Watcher {
           return;
         }
 
-        const added = [];
-        const removed = [];
-        resp.files.forEach((file: Object) => {
-          const normalizedPath = normalizePath(
-            file.name,
-            this.workingDirectory,
-          );
-          if (normalizedPath.indexOf('/node_modules/') !== -1) {
-            return;
-          }
-          if (
-            this.excludes.some((pattern: string): boolean =>
-              minimatch(normalizedPath, pattern))
-          ) {
-            return;
-          }
-          if (file.exists) {
-            added.push({ path: normalizedPath, mtime: +file.mtime_ms });
-          } else {
-            removed.push({ path: normalizedPath });
-          }
-        });
-        if (added.length) {
-          this.onFilesAdded(added);
-        }
-        if (removed.length) {
-          this.onFilesRemoved(removed);
-        }
+        findAllFiles(this.workingDirectory, this.excludes)
```

Comment:
While this may fix an issue for you, it's also going to slow down the file watcher significantly.

Reply:
This is an issue with everyone using `export * from './somePlace'`
I don't think there is any other way to do this.
Also please try it out - I don't notice any slowdown whatsoever.

Reply:
This method is called every time a change is made to a file (either modified, deleted or added). `findAllFiles` will traverse the filesystem, which will be slow on any large project. I think the reason you're not noticing this is that it's being done in the background. 

The `findExports` module already has code to handle `export ... from` declarations. Can you see if you can inject your logic there instead of here?
https://github.com/Galooshi/import-js/blob/a6a7728f9d22c8b0e9fd485c8dfc058dc68c4e0a/lib/findExports.js#L14

Reply:
Say you have a files `a.js`, `b.js`, `c.js`.
The contents of `a.js` are `const abc = someNewVar`
The contents of `b.js` are `export * from './c'`
The contents of `c.js` are `export someOldVar = 123`
When you add `export someNewVar = 321` in file `c.js`
and you try to add the import in `a.js` the only option will be `c.js`
since file `b.js` didn't change. This is an incorrect behavior.
We need to add `b.js` to the `processQueue` method in `ModuleFinder.js`
Since I don't know how to do that - I implemented the only solution I saw

Reply:
Okay, thanks for explaining the use-case. Still, we can't afford traversing through all files on every file modification. I think the way we should solve this is to introduce some kind of dependency list, so that we know to update exports for `b.js` when `c.js` is modified.